src/components/Tweet/index.js

```diff
@@ -0,0 +1,124 @@
+import { AnalyticsIcon } from "../index";
+import { FiMoreVertical } from "icons";
+import { useState, useEffect } from "react";
+import "./index.css";
+import { ADMIN, NOTIFICATIONS } from "config/constants";
+import { deleteTweet, editTweet, getAllBookMarks } from "networkCalls/";
+import { useHome, useNotifyUser, useAuthProvider } from "contexts";
+import { useNavigate } from "react-router-dom";
+import { jwtProfile } from "config/jwt";
+export default function Tweet({
+  post,
+  // setIsTweeted,
+  // setCreateTweet,
+  // setFromEdit,
+}) {
+  const { isLoggedIn } = useAuthProvider();
+  const myProfileDetials = isLoggedIn ? jwtProfile() : null;
+  console.log(myProfileDetials, "buggggggg");
+  const { toast } = useNotifyUser();
+  const navigate = useNavigate();
+  const { setHome } = useHome();
+  const {
+    displayname,
+    content,
+    caption,
+    likes,
+    displayPicture,
+    picture,
+    createdAt,
+    emailId,
+    _id: postid,
+    userId,
+    bookMarked,
+  } = post;
+
+  const [options, setOptions] = useState(false);
+  const [bookMarks, setBookMarks] = useState(null);
+  // deleteHandler
+  const deleteHandler = async () => {
+    try {
+      const deleteResponse = await deleteTweet(postid);
+      setHome({ type: "userTweeted", payload: true });
+      toast.success(NOTIFICATIONS.TWEET_DELETED);
```

Comment:
Suggestion: If using useReducer, you could make an object or a variable for type: "userTweeted" to avoid typo errors for using same type in the dispatch and in the reducer